client/lib/domains/utils.js

```diff
@@ -46,7 +46,7 @@ function getDomainNameFromReceiptOrCart( receipt, cart ) {
 		domainRegistration = cartItems.getDomainRegistrations( cart )[ 0 ];
 	}
 
-	if ( domainRegistration ) {
+	if ( domainRegistration && ( ! receipt || isEmpty( receipt.failed_purchases ) ) ) {
```

Comment:
Why do we need this condition? It seems a little unintuitive to me - this function is supposed to return a domain registration from the receipt or cart, and isn't really concerned failed purchases.